uds/uds_config_tool/FunctionCreation/ReadDataByIdentifierMethodFactory.py

```diff
@@ -100,182 +84,75 @@ def create_requestFunctions(diagServiceElement, xmlElements):
 
                 if semantic == "SERVICE-ID":
-                    responseId = int(param.find("CODED-VALUE").text)
+                    responseId = int(paramElement.find("CODED-VALUE").text)
                     bitLength = int(
-                        (param.find("DIAG-CODED-TYPE")).find("BIT-LENGTH").text
+                        (paramElement.find("DIAG-CODED-TYPE")).find("BIT-LENGTH").text
                     )
-                    listLength = int(bitLength / 8)
-                    responseIdStart = startByte
-                    responseIdEnd = startByte + listLength
-                    totalLength += listLength
-                    SIDLength = listLength
+                    SIDLength = int(bitLength / 8)
                 elif semantic == "ID":
-                    diagnosticId = int(param.find("CODED-VALUE").text)
+                    diagnosticId = int(paramElement.find("CODED-VALUE").text)
                     bitLength = int(
-                        (param.find("DIAG-CODED-TYPE")).find("BIT-LENGTH").text
+                        (paramElement.find("DIAG-CODED-TYPE")).find("BIT-LENGTH").text
                     )
-                    listLength = int(bitLength / 8)
-                    diagnosticIdStart = startByte
-                    diagnosticIdEnd = startByte + listLength
-                    totalLength += listLength
+                    DIDLength = int(bitLength / 8)
                 elif semantic == "DATA":
+                    diagCodedType: DiagCodedType = None
+                    # need to parse the param for the DIAG CODED TYPE
                     dataObjectElement = xmlElements[
-                        (param.find("DOP-REF")).attrib["ID-REF"]
+                        (paramElement.find("DOP-REF")).attrib["ID-REF"]
                     ]
                     if dataObjectElement.tag == "DATA-OBJECT-PROP":
-                        start = int(param.find("BYTE-POSITION").text)
-                        bitLength = int(
-                            dataObjectElement.find("DIAG-CODED-TYPE")
-                            .find("BIT-LENGTH")
-                            .text
-                        )
-                        listLength = int(bitLength / 8)
-                        totalLength += listLength
+                        diagCodedType = getDiagCodedTypeFromDop(dataObjectElement)
+
                     elif dataObjectElement.tag == "STRUCTURE":
-                        start = int(param.find("BYTE-POSITION").text)
-                        listLength = int(dataObjectElement.find("BYTE-SIZE").text)
-                        totalLength += listLength
+                        diagCodedType = getDiagCodedTypeFromStructure(dataObjectElement, xmlElements)
                     else:
+                        # neither DOP nor STRUCTURE
                         pass
+                    param: Param = Param(shortName, bytePosition, diagCodedType)
```

Comment:
why explicitly adding the type info ?

Reply:
removed, was left over from when things were less structured